netbox/dcim/models/devices.py

```diff
@@ -983,6 +983,13 @@ def clean(self):
                 'vc_position': _("A device assigned to a virtual chassis must have its position defined.")
             })
 
+        if hasattr(self, 'vc_master_for') and self.vc_master_for and self.vc_master_for != self.virtual_chassis:
```

Comment:
This can be condensed to something like
```
if vc_master_for := getattr(self, 'vc_master_for') != self.virtual_chassis:
```

Unless `self.virtual_chassis` can be `None`, in which case disregard this

Reply:
Tried a couple different permutations but it was breaking when there was no virtual_chassis / no master, so kept it as is at it works in all cases.